src/audiomixerboard.cpp

```diff
@@ -1092,39 +1092,47 @@ void CAudioMixerBoard::ChangeFaderOrder ( const EChSortType eChSortType )
             iMyFader = static_cast<int> ( i );
         }
 
-        if ( eChSortType == ST_BY_NAME )
+        switch ( eChSortType )
         {
+        case ST_BY_NAME:
             PairList << QPair<QString, size_t> ( vecpChanFader[i]->GetReceivedName().toLower(), i );
-        }
-        else if ( eChSortType == ST_BY_CITY )
-        {
-            PairList << QPair<QString, size_t> ( vecpChanFader[i]->GetReceivedCity().toLower(), i );
-        }
-        else if ( eChSortType == ST_BY_INSTRUMENT )
-        {
+            break;
+        case ST_BY_CITY:
+            // sort first "by city" and second "by name" by adding the name after the city
+            PairList << QPair<QString, size_t> ( vecpChanFader[i]->GetReceivedCity().toLower() + vecpChanFader[i]->GetReceivedName().toLower(), i );
+            break;
+        case ST_BY_INSTRUMENT:
             // sort first "by instrument" and second "by name" by adding the name after the instrument
             PairList << QPair<QString, size_t> ( CInstPictures::GetName ( vecpChanFader[i]->GetReceivedInstrument() ) +
                                                      vecpChanFader[i]->GetReceivedName().toLower(),
                                                  i );
-        }
-        else if ( eChSortType == ST_BY_GROUPID )
-        {
+            break;
+        case ST_BY_GROUPID:
+            // sort first "by group" and second "by name" by adding the name after the group
             if ( vecpChanFader[i]->GetGroupID() == INVALID_INDEX )
             {
                 // put channels without a group at the end
-                PairList << QPair<QString, size_t> ( "z", i ); // group IDs are numbers, use letter to put it at the end
+                PairList << QPair<QString, size_t> ( "999" + vecpChanFader[i]->GetReceivedName().toLower(),
```

Comment:
Seems hacky to offset by 999

Reply:
Far less hacky than using "z" as bigger than "8", though.  The groups will be "001" to "008" and "not a group" will be "999".

Reply:
(And it allows for groups expanding to be one per server channel up to more than max server channels...)